### PR TITLE
104 Inital commit of label studio exporter

### PR DESCRIPTION
**Pull request by Sean1572**
Format our datasets to be easily labeled in label studio
Could make more exporters and make them part of this but small steps

### CODE REVIEW

**Review by kgarwoodsdzwa — COMMENTED** [review]
need to pip install label_studio_sdk

File: data_exporters/label_studio_exporter/demo.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    class_list = ['cluck', 'coocoo',
+                  'twitter', 'alarm', 'chick begging', 'no_buow']
+
+    ds = datasets.Dataset.from_dict({
+        "audio": ls_setup.get_files(
+            ls_file_parent='data/local-files/?d=data1/')["files"],
+        "labels": random.choices(
+            class_list, k=len(
+                ls_setup.get_files(
+                    ls_file_parent='data/local-files/?d=data1/')["files"]
```
Comment (kgarwoodsdzwa) [inline data_exporters/label_studio_exporter/demo.py]: should only need to set this value once

Comment (kgarwoodsdzwa) [reply data_exporters/label_studio_exporter/demo.py]: but it needs to be set in line 25, 38, and here

**Comment by kgarwoodsdzwa** [comment]
works yes but maybe add the pip dependency and then make the labelstudio folder data file in the demo only need 1 user input variable at the beginning rather than needing to change it 3 times depending on where your data is

**Comment by kgarwoodsdzwa** [comment]
ONE ISSUE: the current inference pipeline, as we know, deletes the 'label' column from the resulting output. the logits from prediction are what indicate the actual label. however, in this demo, it uses the label column in the dataset to define the annotation. there needs to be an intermediate step where the inference logits are transformed into labels or this demo needs to have the capability of doing that itself.

**Comment by kgarwoodsdzwa** [comment]
attempting to load data that was inferred upon, having issues because only 1 3 minute audio file was inferred upon, but each second contains a unique label. it doesn't seem to like not being able to assign an id for each audio file, because there is only one audio file but 180 labels, something isnt working. going to keep debugging.

File: data_exporters/label_studio_exporter/demo.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    # labelstudio expects non-int values as predictions, map to labels
+    data['labels'] = [
+        class_list[i] for i in np.argmax(data['pred'], axis=1)
+    ]
+    # audio path must match labelstudio path which is a string
+    for item in data['audio']:
+        item['path'] = Path(item['path']).name
+        for file in file_meta['files']:
+            if item['path'] in file:
+                item['path'] = str(file)
```
Comment (Sean1572) [inline data_exporters/label_studio_exporter/demo.py:59]: Non-breaking change here maybe only keep paths that exist in your dataset? So if we go through all file_meta paths and don't see the item, drop the item from the list (or more safely append items we accept to a new list)

File: data_exporters/label_studio_exporter/demo.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+    # Make sure your file names align to label studio files
+
+    # ===============================================================
+    # below is an example for loading in your inference results
+
+    '''
+    # class list must not be ints
+    # ensure template.xml matches this list
+    class_list = ['your', 'classes']
```
Comment (Sean1572) [inline data_exporters/label_studio_exporter/demo.py:39]: CSE 599, CSE 160, CSE 203b...

**Review by kgarwoodsdzwa — APPROVED** [review]
